### DISCUSSION TITLE
Defaulting `secure` cookie flag to `True`

### DISCUSSION THREAD

**Discussion by giacomomiolo** [encode/starlette#2454] [Potential Issue] [answered]
Hello!

The default at the moment for the [flag secure](https://github.com/encode/starlette/blob/4355e6bc288e14dea9597ad9c9265c0f9cad33c0/starlette/responses.py#L96) is False.

I'm leaning towards the stance of "secure unless explicitly disabled" over the current "insecure unless explicitly enabled". This change would enhance security by ensuring cookies are only sent over HTTPS, but could introduce hurdles in development environments.

Let me know your thoughts on this!

**Comment by Flecart** [comment]
This is a breaking change for services that default to HTTP.
But agree with the security aspect.

**Comment by Kludex** [comment, marked as answer]
The `httponly` flag is already always set on the session cookie. As for `secure`, every major framework (Django, Flask) defaults it to `False` because defaulting to `True` breaks local HTTP development.

I think we can point out this kind of thing in the docs tho.

**Comment by Kludex** [comment]
I've merged this: https://github.com/Kludex/starlette/pull/3169
